File: dart/behavioral/observer/guru_example/bin/concrete_observer_a.dart
import 'concrete_subject.dart';
import 'observer.dart';
import 'subject.dart';

class ConcreteObserverA implements Observer {
  @override
  void update(Subject subject) {
    if (subject is ConcreteSubject &&
        (subject.state < 2 || subject.state > 7)) {
      print('ConcreteObserverA');
    }
  }
}

File: typescript/behavioral/observer/guru_example/concreteObservera.ts
import {Observer} from "./observer";
import {Subject} from "./subject";
import {ConcreteSubject} from "./concreteSubject";

export class ConcreteObserverA implements Observer {
  public update(subject: Subject) {
    if (subject instanceof ConcreteSubject && (subject.state < 3 || subject.state > 5)) {
      console.log('ConcreteObserverA')
    }
  }
}
